docs/root/configuration/http/http_conn_man/route_matching.rst

```diff
@@ -7,8 +7,12 @@ When Envoy matches a route, it uses the following procedure:
 
 #. The HTTP request's *host* or *:authority* header is matched to a :ref:`virtual host
    <envoy_v3_api_msg_config.route.v3.VirtualHost>`.
-#. Each :ref:`route entry <envoy_v3_api_msg_config.route.v3.Route>` in the virtual host is checked,
-   *in order*. If there is a match, the route is used and no further route checks are made.
+#. One of:
+   * Each :ref:`route entry <envoy_v3_api_msg_config.route.v3.Route>` in the virtual host is checked,
+     *in order*. If there is a match, the route is used and no further route checks are made.
+   * The :ref:`matcher entry <envoy_v3_api_msg_.xds.type.matcher.v3.Matcher.MatcherTree>` in the
+     virtual host is used to match a route. If there are many routes this will typically be more
```

Comment:
```suggestion
     virtual host can also be used to match a route. If there are many routes this will typically be more
```

Reply:
this might be a bad suggestion - looking at docs to get some context

Reply:
Yeah, there is no "also" here, only one of the two fields can be set, according to the vhost docs. (Hence "one of" added to the top level bullet.)